packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts

```diff
@@ -128,32 +199,49 @@ export class ConnectionTestDataState {
     this.anonymizeGeneratedData = val;
   }
 
-  *generateTestData(): GeneratorFn<void> {
+  *generateTestDataForDatabaseConnection(serviceExecutionParameters: {
+    query: RawLambda;
+    mapping: Mapping;
+    runtime: Runtime;
+  }): GeneratorFn<void> {
     try {
-      this.generatingTestDataSate.inProgress();
-      const connection = guaranteeNonNullable(
-        this.resolveConnectionValue(this.connectionData.connectionId),
-        `Unable to resolve connection ID '${this.connectionData.connectionId}`,
+      // NOTE: since we don't have a generic mechanism for test-data generation
```

Comment:
this api call takes a bit of time. we should be adding `generatingTestDataSate` to keep track of that and show loading indicator